chapters/en/slides/module4/module4_04.md

```diff
@@ -0,0 +1,131 @@
+
+We can do anything we want:
+    - keep the change made in the local repository,
+    - keep the change made in the remote repository,
+    - write something new to replace both, or get rid of the change entirely.
+
+After your changes the file must be saved, added to the staging area, and then committed before you will be able to push your changes to GitHub.
+
+---
+# Handling merge conflicts: the terminal
+
+
+<center>
+
+<img src='/module4/vc-merge-conflict-t.png' width="70%" alt="404 image"/>
+
+</center>
+
+Notes: It is possible to solve the merge conflicts using the terminal. For this you have to open the editor and find the files where the conflict is.
+You will be able to visualize the merge conflict as appears in the image.
+
+
+
+---
+# Extra: Stashing local non-committed changes before pulling (terminal)
+
+
+```
+error: Your local changes to the following files would be overwritten by merge:
+        README.md
+Please commit your changes or stash them before you merge.
+Aborting
+```
+
+Notes: 
+We have learned that if there are changes on your remote repo in GitHub and you already have local committed changes, you will need to pull before you can push. If the local and remote changes are in the same lines, you will have to resolve the resulting merge conflict, otherwise git will merge automatically. But what if you have just started to make changes to a file when you realize that you forgot to pull before you started to work? The first thing to do is to try to pull, if you're lucky there are either no new changes or they are not in the same file you modified. If they are in the same file, you will get an error message like the one in the slide.
+
+---
+# Extra: Stashing local non-committed changes before pulling (terminal)
+
+`git stash`
+
+
+Notes: If you're about to change some of the files that have also been changed remotely it is a command called `git stash`, which removes your local changes from the working area and and saves them in another location (you can think of this as a secrete pocket which git does not care about when pulling from the remote repo, and from which you can take out the changes again when you need them). You can then do git pull, and follow up with a `git stash pop` to put your changes back from the stash to the working area, and then carry on working.
+
+This workflow can save you from running into merge conflicts, as long as you have not already made modifications to the same lines as you are pulling down. If you have already modified the same file that was updated remotely, you will still run into a merge conflict when you do git stash pop. Stashing is also great when you are working on one feature but realize that you should actually work on another unrelated feature first, you can stash your existing work (instead of manually saving it elsewhere) and finish working on the most urgent feature first.
```

Comment:
```suggestion
This workflow can save you from running into merge conflicts, as long as you have not already made modifications to the same lines as you are pulling down. If you have already modified the same file that was updated remotely, you will still run into a merge conflict when you do `git stash pop`. Stashing is also great when you are working on one feature but realize that you should actually work on another unrelated feature first, you can stash your existing work (instead of manually saving it elsewhere) and finish working on the most urgent feature first.
```